packages/components/badge/src/badge.vue

```diff
@@ -9,8 +9,16 @@
           ns.em('content', type),
           ns.is('fixed', !!$slots.default),
           ns.is('dot', isDot),
+          props.dotClass ?? {},
+        ]"
+        :style="[
+          props.dotStyle ?? {},
+          {
+            backgroundColor: props.color,
+            marginRight: addUnit(-(props.offset?.[0] ?? 0)),
+            marginTop: addUnit(props.offset?.[1] ?? 0),
+          },
```

Comment:
The style seems a bit complicated, maybe we should declare a variable to save it. 🤔